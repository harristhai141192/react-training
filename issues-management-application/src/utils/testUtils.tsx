// Libraries
import React from 'react';
import { ThemeProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

// Themes
import themes from '@themes/index';

const customRender = (component: React.ReactNode) => {
  return render(
    <ThemeProvider theme={themes}>
      <Router>{component}</Router>
    </ThemeProvider>,
  );
};

export * from '@testing-library/react';
export { customRender as render };
