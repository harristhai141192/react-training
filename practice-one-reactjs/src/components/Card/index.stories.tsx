// Libraries
import { ComponentMeta, ComponentStory } from '@storybook/react';

// Components
import Card from '.';

// Images
import { FireDragon } from '@assets/index';

// CSS
import '../../index.css';

export default {
  title: 'Practice-One/Card',
  component: Card,
} as ComponentMeta<typeof Card>;

const Template: ComponentStory<typeof Card> = (args) => <Card {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  cardCode: '012',
  photo: FireDragon,
  name: 'Charmander',
  description: 'Smiling Pokemon Charmander',
};
