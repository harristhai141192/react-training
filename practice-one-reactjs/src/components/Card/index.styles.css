/* Card CSS */
.pokemonCard {
  display: flex;
  flex-direction: row;
  max-width: 350px;
  height: 70px;
  background-color: white;
  padding: 10px;
  box-shadow: 0px 0px 5px 1px lightgrey;
  border-radius: 0.3em;
  margin: 10px;
  overflow: hidden;
}

.pokemonPhoto {
  display: flex;
  flex-grow: 1;
  align-self: center;
  justify-content: center;
}

.pokemonDescription {
  display: flex;
  flex-grow: 2;
  flex-direction: column;
  margin-top: 5px;
}

.pokemonCode {
  color: grey;
}