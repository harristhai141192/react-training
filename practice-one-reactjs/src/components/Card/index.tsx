// Libraries
import React from 'react';

// CSS
import './index.styles.css';

// Components
import Image from '@components/Image';

interface IProps {
  card: {
    code?: string;
    photo?: string;
    name?: string;
    description?: string;
  };
  handleOnClick?: (e: React.MouseEvent<HTMLElement>) => void;
  isOpen?: boolean;
}

const Card: React.FC<IProps> = ({
  card: { code = '', photo = '', name = '', description = '' },
  handleOnClick,
}) => (
  <div className='pokemonCard' onClick={kkaasd()}>
    <div className='pokemonPhoto'>
      <Image source={photo} alt={description} width='50px' />
    </div>
    <div className='pokemonDescription'>
      <div className='pokemonCode'>{code}</div>
      <div className='pokemonName'>{name}</div>
    </div>
  </div>
);

export default Card;
