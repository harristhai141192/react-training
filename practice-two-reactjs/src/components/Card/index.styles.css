.cardDetail {
  display: flex;
  flex-direction: column;
  text-align: left;
  padding: 20px;
  justify-content: end;
  height: 100%;
  background-image: linear-gradient(to bottom, rgba(230, 230, 230, 0), rgba(30, 29, 29, 0.459));
  color: white;
}

