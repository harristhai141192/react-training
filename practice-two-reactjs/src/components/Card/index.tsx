import { Box, Text } from '@chakra-ui/react';

import { Member } from '@models/index';
import { Link } from 'react-router-dom';

interface IProps {
  card: Member;
  margin?: string;
  height?: { sm?: string; md?: string; lg?: string };
  onClick?: () => void;
  linkToPage?: string;
}

const Card = ({ margin, height, onClick, card, linkToPage = '' }: IProps) => {
  return (
    <Box
      h={height}
      w={{ sm: '100%', md: '25%', lg: '25%' }}
      backgroundImage={card.memberImg}
      backgroundPosition='center'
      backgroundRepeat='no-repeat'
      backgroundSize='cover'
      data-testid='testCard'
      margin={margin}
      onClick={onClick}
    >
      <Link to={linkToPage}>
        <Box
          display='flex'
          flexDirection='column'
          textAlign='left'
          p='20px'
          justifyContent='end'
          height='100%'
          backgroundImage='linear-gradient(to bottom, rgba(230, 230, 230, 0), rgba(30, 29, 29, 0.459))'
          color='white'
        >
          <Text
            className='textTitle'
            fontFamily='RalewayExtraBold'
            fontSize={{ sm: 'text.extraLarge', md: 'text.large', lg: 'text.extraLarge' }}
          >
            {card.memberName}
          </Text>
          <Text className='textDescription' fontFamily='RalewayLight' fontSize='text.small'>
            Member Since {card.memberSince?.substring(0, 4)}
          </Text>
        </Box>
      </Link>
    </Box>
  );
};

export default Card;
