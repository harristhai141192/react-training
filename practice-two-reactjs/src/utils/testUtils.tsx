import { ThemeProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import themes from '@themes/index';
import { ReactNode } from 'react';

const customRender = (component: ReactNode) => {
  return render(<ThemeProvider theme={themes}>{component}</ThemeProvider>);
};

export * from '@testing-library/react';
export { customRender as render };
