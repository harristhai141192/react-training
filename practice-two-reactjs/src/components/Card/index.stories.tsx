// Libs
import { Box } from '@chakra-ui/react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { withRouter } from 'storybook-addon-react-router-v6';

// Component
import Card from '.';

export default {
  title: 'Components/Card',
  component: Card,
  decorators: [withRouter],
  argTypes: {},
} as ComponentMeta<typeof Card>;

const Template: ComponentStory<typeof Card> = (args) => (
  <Box width='1000px'>
    <Card {...args} />
  </Box>
);

export const Default = Template.bind({});
Default.args = {
  card: {
    memberImg: "url('src/assets/images/Frankie.svg')",
    memberSince: '1996/14/11',
    memberName: 'Frankie',
  },
  linkToPage: '',
};
