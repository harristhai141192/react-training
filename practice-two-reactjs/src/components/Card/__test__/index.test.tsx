// libraries
import { getByTestId, getByText, render } from '@utils/testUtils';
// components
import Card from '../index';

// theme

describe('Card component', () => {
  test('Component [Card] should match snapshot', () => {
    const { container } = render(<Card />);

    expect(container).toMatchSnapshot();
  });

  test('Component [Card] should render correctly with subText, titleText and Image', () => {
    const { container } = render(
      <Card
        imageBg="url('src/assets/images/Frankie.svg')"
        subText='Member since 2016'
        titleText='Frankie'
        width='350px'
      />,
    );

    const titleText = getByText(container, 'Frankie');
    const subText = getByText(container, 'Member since 2016');
    const getBgImg = getByTestId(container, 'testCard');

    expect(titleText).toBeTruthy;
    expect(subText).toBeTruthy;
    expect(getBgImg.style.backgroundImage).toBe('');
  });
});
