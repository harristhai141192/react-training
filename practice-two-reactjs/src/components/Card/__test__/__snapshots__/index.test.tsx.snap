// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Component [Card] should match snapshot 1`] = `
<div>
  <div
    class="css-17r06r5"
    data-testid="testCard"
  >
    <div
      class="css-14xodac"
    >
      <p
        class="chakra-text textTitle css-1bvh0q3"
      />
      <p
        class="chakra-text textDescription css-sm1wli"
      />
    </div>
  </div>
</div>
`;
